### PR TITLE
Add support for absolute paths for jacoco exec and report

### PR DESCRIPTION
**Pull request by vsevel**
This changes the `data-file` and `report-location` config parameters to allow paths that are not relative to the build directory.
we need this to be able to point to a common location for all modules in a multi module project.
today, if we specify a `data-file=jacoco/jacoco-quarkus.exec` in module foo, the `data-file` passed to jacoco will be `<project root>/foo/target/jacoco/jacoco-quarkus.exec`
in a multi-module project, we need the path to be something like <project root>/target/jacoco/jacoco-quarkus.exec
the only option we have right now is to use an expression such as `data-file=../target/jacoco/jacoco-quarkus.exec`.
but that works only for modules that are at the first level. if we have deeper sub-modules (which is the case in extensions), then we have to customize the number of `../../..` for each module.
This PR allows to configure `data-file=${maven.multiModuleProjectDirectory}/target/jacoco/jacoco-quarkus.exec`

this is a breaking change on the configuration, since existing configuration `data-file=foo` would end up with `./foo` instead of `./target/foo`. if accepted this will need to be properly documented in the migration guide.
another option would be to add an additional config parameter `relative-to-multi-module-project-directory=true|false(false by default)`.

see [zulip Multi Module Code Coverage With Jacoco](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Multi.20Module.20Code.20Coverage.20With.20Jacoco) and https://github.com/quarkusio/quarkus/issues/32254

### CODE REVIEW

File: test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java [hunk 1]
```diff
@@ -11,10 +11,11 @@
...
 
     /**
-     * The jacoco data file
+     * The jacoco data file. By default this will be target/jacoco-quarkus.exec.
+     * The path can be relative (to the module) or absolute.
      */
-    @ConfigItem(defaultValue = "jacoco-quarkus.exec")
-    public String dataFile;
+    @ConfigItem
+    public Optional<String> dataFile;
```
Comment (geoand) [inline test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:18]: What's the rationale of this change?

File: test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java [hunk 2]
```diff
@@ -82,8 +83,9 @@ public class JacocoConfig {
...
 
     /**
-     * The location of the report files.
+     * The location of the report files. By default this will be target/jacoco-report.
+     * The path can be relative (to the module) or absolute.
      */
-    @ConfigItem(defaultValue = "jacoco-report")
-    public String reportLocation;
+    @ConfigItem
+    public Optional<String> reportLocation;
```
Comment (geoand) [inline test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:90]: Same here

**Review by geoand — COMMENTED** [review]
Thanks for this!

I think it makes sense, I just have a comment about the config object change

File: test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java (hunk 1 above)
Comment (vsevel) [reply test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:18]: I changed the default to be the equivalent of `target/jacoco-quarkus.exec`, but since I wanted to reuse `outputTargetBuildItem.getOutputDirectory()` and `module.getWorkspaceModule().getBuildDir()` in the processor, I could not use a `WithDefault` anymore.
so now it is optional, but a default is calculated in the processor directly.
there are no changes for people who did not set `data-file`.
for people that did set `data-file=foo`, they need to change it to `target/foo`.
your advice more than welcome.

Comment (geoand) [reply test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:18]: Makes sense, thanks.

What I would like however is some Javadoc explaining this process so users who might want to configure it can find this information on our config page (or their IDE if it supports it)

Comment (geoand) [reply test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:18]: I guess it's fine.

**Review by geoand — APPROVED** [review]

File: test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java [hunk 3]
```diff
@@ -11,10 +11,11 @@
 public class JacocoConfig {
 
     /**
-     * The jacoco data file
+     * The jacoco data file. By default this will be target/jacoco-quarkus.exec.
```
Comment (gsmet) [inline test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:14]: Could we use `defaultValueForDocumentation` instead?

Comment (geoand) [reply test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:14]: I didn't remember that at all. @vsevel fancy doing a new PR with this improvement?

Comment (vsevel) [reply test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java:14]: Sure. I will add some info on the guide as well
